tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -1745,4 +1745,27 @@ public function testMergeWithCreate()
         $this->assertNotEmpty($result->errors('author_id'));
         $this->assertNotEmpty($result->errors('thing'));
     }
+
+    /**
+     * Test Model.beforeMarshal event.
+     *
+     * @return void
+     */
+    public function testBeforeMarshalEvent()
+    {
+        $data = [
+            'title' => 'My title',
+            'body' => 'My content'
+        ];
+
+        $marshall = new Marshaller($this->articles);
+
+        $this->articles->eventManager()->attach(function ($e, $data) {
+            $data['title'] = 'Modified title';
```

Comment:
Basically what we wanted to see in the test is that the beforeMarshal event is triggered for the Users table as well